xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/xwiki.js

```diff
@@ -1821,15 +1821,34 @@ document.observe("xwiki:dom:loaded", function() {
  * Intercept asynchronous HTTP requests and look for the custom X-XWIKI-HTML-HEAD and X-XWIKI-HTML-SCRIPTS response
  * headers that are used by the server to extend the page HTML head asynchronously.
  */
-require(['jquery'], function($) {
+ (function() {
   /**
    * Collect the URLs of all the external stylesheets an JavaScript files that are already loaded (or in the process of
    * being loaded) because their associated tag is in the DOM.
    */
   var getLoadedResources = function() {
-    return $('link[href], script[src]').map(function() {
-      return $(this).attr('href') || $(this).attr('src');
-    }).get();
+    var resources = [];
+    [...document.querySelectorAll('link[href], script[src]')].forEach(function(resource) {
+      resources.push(resource.getAttribute('href') || resource.getAttribute('src'))
+    });
+    return resources;
+  };
+
+  /**
+   * Injects the given stylesheet or JavaScript file into the page head.
+   *
+   * @param {Element} resource stylesheet or JavaScript resource
+   */
+  var loadResource = function(resource) {
+    if (resource.getAttribute('src')) {
+      let script = document.createElement("script");
+      script.setAttribute('defer', '');
+      script.setAttribute('type', 'text/javascript');
+      script.setAttribute('src', resource.getAttribute('src'));
```

Comment:
Have you tried cloning the script element instead. My worry is that you lose some of the attributes from the script element which could be important (the script may read them with https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript ). In any case it needs a comment explaining why we treat ``script`` and ``link`` tags differently.

Reply:
No, it's not working with clone either. I'll add a comment
Another way is using `document.createRange().createContextualFragment(resource.outerHTML)`, to preserve the script attributes (see https://ghinda.net/article/script-tags/)

Reply:
Couldn't we use this in ``extendPageHead`` to convert the string HTML to a ``DocumentFragment`` from where to extract the ``script`` and ``link`` tags, instead of using the detached ``DIV`` element? It feels redundant to go from string to DOM (using the ``DIV``) and then back to string (``outerHTML``) and finally back to DOM (``createContextualFragment``), when we could go directly from string to DOM using ``createContextualFragment`` (using the ``HEAD`` element as context).